lib/capability/capability.go

```diff
@@ -47,35 +52,43 @@ func GetCapabilitiesStatus(capabilities ClusterCapabilities) configv1.ClusterVer
 	return status
 }
 
-// CheckResourceEnablement, given resource annotations and defined cluster capabilities, checks if the capability
-// annotation exists. If so, each capability name is validated against the known set of capabilities. Each valid
-// capability is then checked if it is disabled. If any invalid capabilities are found an error is returned listing
-// all invalid capabilities. Otherwise, if any disabled capabilities are found an error is returned listing all
-// disabled capabilities.
-func CheckResourceEnablement(annotations map[string]string, capabilities ClusterCapabilities) error {
+// GetResourceCapabilities, given resource annotations and defined cluster capabilities, checks if the capability
+// annotation exists. If so, each capability name is validated against the known set of capabilities. If any
+// invalid capabilities are found an error is returned listing all invalid capabilities. Otherwise, all
+// resource capabilities are returned.
+func GetResourceCapabilities(annotations map[string]string, capabilities ClusterCapabilities) (error, []configv1.ClusterVersionCapability) {
 	val, ok := annotations[CapabilityAnnotation]
 	if !ok {
-		return nil
+		return nil, nil
 	}
 	caps := strings.Split(val, "+")
 	numCaps := len(caps)
 	unknownCaps := make([]string, 0, numCaps)
-	disabledCaps := make([]string, 0, numCaps)
+	allCaps := make([]configv1.ClusterVersionCapability, 0, numCaps)
 
 	for _, c := range caps {
 		if _, ok = capabilities.KnownCapabilities[configv1.ClusterVersionCapability(c)]; !ok {
 			unknownCaps = append(unknownCaps, c)
-		} else if _, ok = capabilities.EnabledCapabilities[configv1.ClusterVersionCapability(c)]; !ok {
-			disabledCaps = append(disabledCaps, c)
+		} else {
+			allCaps = append(allCaps, configv1.ClusterVersionCapability(c))
 		}
 	}
 	if len(unknownCaps) > 0 {
-		return fmt.Errorf("unrecognized capability names: %s", strings.Join(unknownCaps, ", "))
+		return fmt.Errorf("unrecognized capability names: %s", strings.Join(unknownCaps, ", ")), nil
 	}
-	if len(disabledCaps) > 0 {
-		return fmt.Errorf("disabled capabilities: %s", strings.Join(disabledCaps, ", "))
+	return nil, allCaps
+}
+
+// ResourceEnabled returns whether given resource capabilities indicates resource is enabled.
+func ResourceEnabled(resourceCapabilities []configv1.ClusterVersionCapability,
+	enabledCapabilities map[configv1.ClusterVersionCapability]struct{}) bool {
+
+	for _, c := range resourceCapabilities {
+		if _, ok := enabledCapabilities[c]; !ok {
+			return false
+		}
 	}
-	return nil
+	return true
 }
 
 // setKnownCapabilities populates a map keyed by capability from all known capabilities as defined in ClusterVersion.
```

Comment:
orthogonal to this PR, but I'm realizing that `setKnownCapabilities` doesn't need its current `config *configv1.ClusterVersion` argument.